flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGTest.java

```diff
@@ -0,0 +1,171 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.executiongraph;
+
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.metrics.groups.UnregisteredMetricsGroup;
+import org.apache.flink.runtime.JobException;
+import org.apache.flink.runtime.blob.VoidBlobWriter;
+import org.apache.flink.runtime.checkpoint.StandaloneCheckpointRecoveryFactory;
+import org.apache.flink.runtime.client.JobExecutionException;
+import org.apache.flink.runtime.executiongraph.failover.AdaptedRestartPipelinedRegionStrategyNG;
+import org.apache.flink.runtime.executiongraph.failover.FailoverStrategyLoader;
+import org.apache.flink.runtime.executiongraph.failover.PipelinedFailoverRegionBuildingTest;
+import org.apache.flink.runtime.executiongraph.failover.flip1.RestartPipelinedRegionStrategyBuildingTest;
+import org.apache.flink.runtime.executiongraph.restart.NoRestartStrategy;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.runtime.jobmaster.slotpool.SlotProvider;
+import org.apache.flink.runtime.shuffle.NettyShuffleMaster;
+import org.apache.flink.runtime.testingUtils.TestingUtils;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for {@link AdaptedRestartPipelinedRegionStrategyNG}.
+ */
+public class AdaptedRestartPipelinedRegionStrategyNGTest extends TestLogger {
+
+	/**
+	 * Test that {@link AdaptedRestartPipelinedRegionStrategyNG} is loaded
+	 * as failover strategy if {@link JobManagerOptions#EXECUTION_FAILOVER_STRATEGY}
+	 * value is "region".
+	 */
+	@Test
+	public void testAdaptedRestartPipelinedRegionStrategyNGLoading() throws Exception {
+		final ExecutionGraph eg = createExecutionGraph(new JobGraph());
+
+		assertTrue(eg.getFailoverStrategy() instanceof AdaptedRestartPipelinedRegionStrategyNG);
+	}
+
+	/**
+	 * Test whether region building works. This helps to make sure that the
+	 * overall region building process is not breaking.
+	 * More detailed region building verification is covered
+	 * in {@link RestartPipelinedRegionStrategyBuildingTest}
+	 * <pre>
+	 *     (v11) --> (v21) -+-> (v31) --> (v41)
+	 *                      x
+	 *     (v12) --> (v22) -+-> (v32) --> (v42)
+	 *
+	 *                      ^
+	 *                      |
+	 *                  (blocking)
+	 * </pre>
+	 * 4 regions. Each region has 2 pipelined connected vertices.
+	 */
+	@Test
+	public void testRegionBuilding() throws Exception {
```

Comment:
This test doesn't add code coverage because we are redunantly unit testing the behavior of `RestartPipelinedRegionStrategy`. We should test the behavior that is added by `AdaptedRestartPipelinedRegionStrategyNG`.

I think we can delete it together with the method `AdaptedRestartPipelinedRegionStrategyNG#getFailoverRegion()`.

Reply:
Ok.